parsers/ERP_PGCB.py

```diff
@@ -68,47 +71,42 @@ def parse_table_body(table_body: Tag):
         row_items = row.find_all("td")
         row_items = [item.text.strip() for item in row_items]
 
-        row_dict = dict()
-
         # date and time in [0]; [1] are DD-MM-YYYY; HH:mm[:ss]
-        parsed_day = arrow.get(row_items[0], "DD-MM-YYYY", tzinfo=tz).datetime
+        parsed_day = datetime.strptime(row_items[0], "%d-%m-%Y").date()
         try:
-            parsed_time = arrow.get(row_items[1], "HH:mm:ss", tzinfo=tz).time()
-        except arrow.parser.ParserMatchError:
+            assert isinstance(row_items[1], str)
+            if row_items[1][0:2] == "24":
+                # The endpoint is reporting 24:00:00 as 00:00:00 of the next day, so we need to fix that.
+                row_items[1] = row_items[1].replace("24", "00", 1)
+                parsed_day += timedelta(days=1)
+            # newer data points are in HH:mm:ss format
+            parsed_time = datetime.strptime(row_items[1], "%H:%M:%S").time()
+        except ValueError:
             # very old data points are sometimes in HH:mm format
-            parsed_time = arrow.get(row_items[1], "HH:mm", tzinfo=tz).time()
-
-        row_dict["time"] = datetime.combine(parsed_day, parsed_time, tzinfo=tz)
-
-        # [2] is total generation in MW
-        row_dict["total_generation"] = table_entry_to_float(row_items[2])
-        # [3] is total demand in MW
-        row_dict["total_demand"] = table_entry_to_float(row_items[3])
-        # [4] is loadshed, not interesting for us
-        row_dict["loadshed"] = table_entry_to_float(row_items[4])
-        # [5] is generation from gas
-        row_dict["gas"] = table_entry_to_float(row_items[5])
-        # [6] is generation from liquid fuel (oil)
-        row_dict["oil"] = table_entry_to_float(row_items[6])
-        # [7] is generation from coal
-        row_dict["coal"] = table_entry_to_float(row_items[7])
-        # [8] is generation from hydro
-        row_dict["hydro"] = table_entry_to_float(row_items[8])
-        # [9] is generation from solar
-        row_dict["solar"] = table_entry_to_float(row_items[9])
-        # [10] is generation from wind
-        row_dict["wind"] = table_entry_to_float(row_items[10])
-        # [11], [12] is import from Bheramara PP (IN-EA) and Tripura (IN-NE)
-        row_dict["bd_import_bheramara"] = table_entry_to_float(row_items[11])
-        row_dict["bd_import_tripura"] = table_entry_to_float(row_items[12])
-        row_dict["remarks"] = row_items[13]
-
-        row_data.append(row_dict)
+            parsed_time = datetime.strptime(row_items[1], "%H:%M").time()
+
+        row_data.append(
+            {
+                "time": datetime.combine(parsed_day, parsed_time, tzinfo=TIMEZONE),
+                "total_generation": table_entry_to_float(row_items[2]),  # MW
```

Comment:
Shall we also add a parser function for this?
I know it's not really used by us right now but would be nice to start collecting these to use in validation rules in the future?